src/nvim/window.c

```diff
@@ -3653,7 +3652,8 @@ static win_T *win_alloc(win_T *after, int hidden)
     win_append(after, new_wp);
 
   new_wp->w_wincol = 0;
-  new_wp->w_width = Columns;
+  assert(Columns <= INT_MAX);
+  new_wp->w_width = (int)Columns;
```

Comment:
I'm starting to wonder if we shouldn't just have a `long_to_int()` that does this assert automatically.